### DISCUSSION TITLE
PineconeAuthorizationError

### DISCUSSION THREAD

**Discussion by YannLA69** [FlowiseAI/Flowise#2155] [Q&A]
Hello everyone,

I'm trying to use Pinecone but I keep getting an error: "Error: PineconeAuthorizationError: The API key provided was rejected while calling https://api.pinecone.io/indexes/rlpyon. Please check your configuration values and try again. You can find the configuration values for your project in the Pinecone developer console at https://app.pinecone.io"

I checked the Pinecone side with customer service and everything seems to be ok. I saw that several people had this problem but I didn't see any response :(

I'm using a paid version of Pinecone so that's not the problem.
Of course, I checked and even changed the API key, and checked the index.
I wonder if the problem isn't with the request url, but I can't see how to change it to test.

Thanks for any help you can give me! :)
![Capture d’écran 2024-04-11 à 09 37 54](https://github.com/FlowiseAI/Flowise/assets/155621839/cb527bef-1651-40fe-b61a-8300807964dc)

**Comment by YannLA69** [comment]
I think I'm going crazy ^^'
I'm really not getting anywhere and I don't understand why at all. I know I'm a novice, but still...

I put Pinecone aside to test with an "in memory vector store", but even there I get an error: "memory.getChatMessages is not a function".
It seems there's a logical flowise function I'm missing. Does anyone have any clues?

Thanks a lot!
![Capture d’écran 2024-04-15 à 10 55 59](https://github.com/FlowiseAI/Flowise/assets/155621839/b366ecaa-b3f6-470f-970b-4526913f4e8e)

**Comment by YannLA69** [comment]
After weeks of searching, my problem was solved!

For the "in memory vector store", I just hadn't upsert the documents... 🙄
Once the upsert went well, I quit then relaunched (to clear the memory), then the Pinecone upsert worked. Why did this happen? No idea, but at least I can move on.
If anyone has a possible explanation for this phenomenon, I'd love to hear it, as I like to understand what's going on.

**Comment by mattarredona** [comment]
Hello friend! I have the exact same problem as you! I'm still a beginner and I did the API integration and indexes correctly, but the error still persists! This drives me crazy, because nothing makes sense... I've already deleted everything, created a new account on Pinecone, redid the whole process and nothing. It's very strange that it worked normally, but suddenly it stopped working and the error persists. This drives me crazy, because nothing makes sense...
